src/Microsoft.AspNet.Mvc.Razor/RazorPage.cs

```diff
@@ -618,9 +636,13 @@ public virtual void WriteLiteralTo([NotNull] TextWriter writer, string value)
                     WriteUnprefixedAttributeValueTo(valueBuffer, value);
                 }
 
-                var htmlString = new HtmlString(valueBuffer.ToString());
+                using (var stringWriter = new StringWriter())
+                {
+                    valueBuffer.Content.WriteTo(stringWriter, HtmlEncoder);
 
-                executionContext.AddHtmlAttribute(attributeName, htmlString);
+                    var htmlString = new HtmlString(stringWriter.ToString());
+                    executionContext.AddHtmlAttribute(attributeName, htmlString);
+                }
```

Comment:
What goes South if you instead call `executionContext.AddHtmlAttribute(attributeName, valueBuffer.Content)`?